### PR TITLE
Add art.taunoerik.TaunoMonitor

### PR DESCRIPTION
**Pull request by taunoe**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build art.taunoerik.TaunoMonitor

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/art.taunoerik.TaunoMonitor.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
• tag-missing           : <url> is not present
Validation of files failed
program finished with exit code 1
```

File: art.taunoerik.TaunoMonitor.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
                 {
                     "type" : "git",
                     "tag"  : "v0.1.0",
-                    "commit" : "7ed784d9cda2ed58d14c5e728875b596800d902b",
+                    "commit" : "632a158da0416902e0c287c75bb45ca69af5b0ac",
```
Comment (hfiguiere) [inline art.taunoerik.TaunoMonitor.json:48]: retagging is bad.

**Comment by hfiguiere** [comment]
bot, build art.taunoerik.TaunoMonitor

File: art.taunoerik.TaunoMonitor.json (hunk 1 above)
Comment (hfiguiere) [reply art.taunoerik.TaunoMonitor.json:48]: or it is not, and in that case that's why the build failed.

File: art.taunoerik.TaunoMonitor.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    	],
+    	"sources": [
+            {
+              "type": "file",
+              "url": "https://files.pythonhosted.org/packages/07/bc/587a445451b253b285629263eb51c2d8e9bcea4fc97826266d186f96f558/pyserial-3.5-py2.py3-none-any.whl",
+              "sha256": "c4451db6ba391ca6ca299fb3ec7bae67a5c55dde170964c7a14ceefec02f2cf0"
+            }
+    	  ]
+	},
+        {
```
Comment (bbhtt) [inline art.taunoerik.TaunoMonitor.json:40]: Can you fix the indentation here?

**Comment by taunoe** [comment]
Is it a problem that I changed the app id to lowercase?

**Comment by bbhtt** [comment]
> Is it a problem that I changed the app id to lowercase?

No, it's fine if you're fine and made the appropriate changes. The spec recommends lowercased id.

**Comment by bbhtt** [comment]
bot, build art.taunoerik.tauno-monitor

**Comment by bbhtt** [comment]
You didn't fix these https://github.com/flathub/flathub/pull/4711#issuecomment-1817906872

```
builddir/files/share/appdata/art.taunoerik.tauno-monitor.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
• tag-missing           : <url> is not present
```

**Comment by hfiguiere** [comment]
these are the same as https://github.com/flathub/flathub/pull/4711#issuecomment-1817906872

**Comment by taunoe** [comment]
> You didn't fix these [#4711 (comment)](https://github.com/flathub/flathub/pull/4711#issuecomment-1817906872)
> 
> ```
> builddir/files/share/appdata/art.taunoerik.tauno-monitor.appdata.xml: FAILED:
> • tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
> • style-invalid         : Not enough <screenshot> tags, minimum is 1
> • tag-missing           : <release> required
> • tag-missing           : <name> is not present
> • tag-missing           : <summary> is not present
> • tag-missing           : <url> is not present
> ```

I added [art.taunoerik.tauno-monitor.appdata.xml](https://github.com/taunoe/tauno-monitor/blob/main/data/art.taunoerik.tauno-monitor.appdata.xml) file. Wrong location?

**Comment by hfiguiere** [comment]
it's an error to have both .appdata.xml and .metainfo.xml

**Comment by bbhtt** [comment]
bot, build art.taunoerik.tauno-monitor

**Comment by bbhtt** [comment]
```
Invalid desktop filename: art.taunoerik.tauno-monitor.desktop.in
Error: ERROR: appstream-compose failed: Child process exited with code 1
```

**Comment by hfiguiere** [comment]
> ```
> Invalid desktop filename: art.taunoerik.tauno-monitor.desktop.in
> Error: ERROR: appstream-compose failed: Child process exited with code 1
> ```

This https://github.com/taunoe/tauno-monitor/blob/v0.1.2/data/art.taunoerik.tauno-monitor.metainfo.xml.in#L14

is definitely wrong

**Comment by taunoe** [comment]
Removed the '.in'

**Comment by bbhtt** [comment]
bot, build art.taunoerik.tauno-monitor

**Comment by bbhtt** [comment]
> Build [83244 successful](https://buildbot.flathub.org/#/builders/6/builds/83244) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/65878/art.taunoerik.tauno-monitor.flatpakref
> ```

Clicking the open button shows this:

```
Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/serial/serialposix.py", line 322, in open
    self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/dev/ttyS0'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/share/tauno-monitor/tauno_monitor/window.py", line 141, in btn_open
    self.tauno_serial.open(selected_port, selected_baud_rate)
  File "/app/share/tauno-monitor/tauno_monitor/window.py", line 198, in open
    self.myserial.open()
  File "/app/lib/python3.11/site-packages/serial/serialposix.py", line 325, in open
    raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg))
serial.serialutil.SerialException: [Errno 13] could not open port /dev/ttyS0: [Errno 13] Permission denied: '/dev/ttyS0'
```

**Comment by bbhtt** [comment]
Hrm looks like `usermod -a -G dialout $USER` does not work on fedora https://discussion.fedoraproject.org/t/how-to-add-myself-to-the-dialout-group/24147 Needs to be manually added or /dev/ttyS0 chowned to $USER.

You should probably mention this somewhere, typically a regular user is not part of the dialout group on any distro afaik.

Then atleast open doesn't show the error and shows this:

```
Btn Open
Tauno Serial Open()
Serial is open
```

**Comment by taunoe** [comment]
I will add this info to the description.

**Comment by hfiguiere** [comment]
> I will add this info to the description.

it should display a use visible error. that the least that can be done.

**Comment by taunoe** [comment]
I added banner and a short guide to the app

Screenshot from 2023-11-25 20-14-53

Screenshot from 2023-11-25 20-15-05

**Comment by bbhtt** [comment]
bot, build art.taunoerik.tauno-monitor

**Pull request closed without merging**